osquery/events/CMakeLists.txt

```diff
@@ -27,6 +27,7 @@ endif()
 file(GLOB OSQUERY_KERNEL_QUEUE "kernel/*.cpp")
 ADD_OSQUERY_LIBRARY(TRUE osquery_events
   events.cpp
+  kernel.cpp
```

Comment:
Let's write this block (and the test block below) like:

```cmake
ADD_OSQUERY_LIBRARY(TRUE osquery_events
  events.cpp
)

file(GLOB OSQUERY_KERNEL_QUEUE "kernel/*.cpp")
ADD_OSQUERY_LIBRARY(FALSE osquery_additional_events
  kernel.cpp
  ${OSQUERY_KERNEL_QUEUE}
)

file(GLOB OSQUERY_KERNEL_QUEUE_TEST "kernel/tests/*.cpp")
ADD_OSQUERY_TEST(FALSE ${OSQUERY_KERNEL_QUEUE_TEST})

```

The first parameter to these CMake macros determines if the code should be included in the osquery SDK. We don't want the kernel code in `libosquery.a` and the test should only build if you're testing osqueryi/osqueryd targets, not extensions/modules linked using the development libs.